### PR TITLE
Downloader improvements

### PR DESCRIPTION
**Pull request by tagoylo**
Added function to get git sha properties of files.
Added check if file exists in artifactory.

### CODE REVIEW

File: nebula/downloader.py [hunk 1]
```diff
@@ -62,9 +64,63 @@ def get_newest_folder(links):
 
     return dates[-1]
 
+def get_gitsha(branch, link, daily=False):
+    server = "artifactory.analog.com"
+    dest = "outs"
+    if not os.path.isdir(dest):
+        os.mkdir(dest)
+    file = os.path.join(dest, "properties.yaml")
+    f = open(file, "w")
```
Comment (kimpaller) [inline nebula/downloader.py:73]: Maybe inclose block 73- 121 with 'with' clause instead to make sure program closes file incase unexpected failure:
ex.
with open(file, w) as f:
     \<code block goes here\>

File: nebula/downloader.py [hunk 2]
```diff
@@ -62,9 +64,63 @@ def get_newest_folder(links):
...
+        else:
+            url = link.format(server, "", "", "")
+            release_folder = get_latest_release(listFD(url))
+            url = link.format(server, release_folder, "", "")
+            folder = get_newest_folder(listFD(url[:-1]))
+        url = url +"/"+str(folder)
+        path = ArtifactoryPath(url)
+        bootpartition = {"bootpartition_folder": ntpath.basename(path), "bootpartition_git_props": path.properties}
+        yaml.dump(bootpartition, f)
+        f.close()
```
Comment (kimpaller) [inline nebula/downloader.py:89]: if 'with' clause is implemented, there is no need to explicitly close the file.

File: nebula/downloader.py [hunk 3]
```diff
@@ -62,9 +64,63 @@ def get_newest_folder(links):
...
+            url_hdl = url +"/"+str(folder)
+
+        path_linux = ArtifactoryPath(url_linux)
+        path_hdl = ArtifactoryPath(url_hdl)
+        linux_props = {"linux_folder": ntpath.basename(path_linux), "linux_git_props": path_linux.properties}
+        hdl_props = {"hdl_folder": ntpath.basename(path_hdl), "hdl_git_props": path_hdl.properties}
+        properties = linux_props.copy()
+        properties.update(hdl_props)
+        yaml.dump(properties, f)
+        f.close()
```
Comment (kimpaller) [inline nebula/downloader.py:124]: if 'with' clause is implemented, there is no need to explicitly close the file.

**Review by kimpaller — COMMENTED** [review]
Aside from the 'wiith' clause implementation, code seems ok and captures fairly well the requirement.

File: nebula/downloader.py (hunk 1 above)
Comment (tagoylo) [reply nebula/downloader.py]: Implemented through this commit 115e661.

File: nebula/downloader.py (hunk 2 above)
Comment (tagoylo) [reply nebula/downloader.py]: Implemented through this commit 115e661.

**Review by kimpaller — APPROVED** [review]

File: nebula/downloader.py (hunk 3 above)
Comment (tagoylo) [reply nebula/downloader.py]: Implemented through this commit 115e661.

**Pull request merged**